src/main/scala/chiseltest/backends/VPIInterface.scala

```diff
@@ -0,0 +1,427 @@
+package chiseltest.backends
+
+import java.io.File
+import java.nio.channels.FileChannel
+
+import chisel3._
+import firrtl.ir.{GroundType, IntWidth, Port}
+import logger.LazyLogging
+
+import scala.collection.immutable.ListMap
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+import scala.concurrent.duration._
+import scala.concurrent.{blocking, Await, ExecutionContext, Future}
+import scala.language.implicitConversions
+import scala.sys.process.{Process, ProcessLogger}
+
+class VPIInterface(topPorts: Seq[Port], topName: String, commands: Seq[String])
+    extends SimulatorInterface
+    with LazyLogging {
+  def cycleCount: BigInt = _currentClock
+
+  def poke(signal: String, value: BigInt): Unit = {
+    if (inputsNameToChunkSizeMap contains signal) {
+      _pokeMap(signal) = value
+      isStale = true
+    } else {
+      val id = _signalMap.getOrElseUpdate(signal, getId(signal))
+      if (id >= 0) {
+        poke(id, _chunks.getOrElseUpdate(signal, getChunk(id)), value)
+        isStale = true
+      } else {
+        logger.info(s"Can't find $signal in the emulator...")
```

Comment:
This should be a hard failure (and similarly below)

Reply:
I'll add an additional API to enable/disable this.
In the future, it's possible to peek a non-exist(constant-propagated) target inside Module.